pkg/agent/entrypoint.go

```diff
@@ -70,25 +78,15 @@ func (r MainRunner) Run(ctx context.Context) error {
 
 	klog.Info("Entering main loop")
 	for {
-		select {
-		case <-ctx.Done():
-			vmWatchStore.Stop()
-			schedulerStore.Stop()
-
-			// Remove anything else from vmEvents
-		loop:
-			for {
-				select {
-				case <-vmEvents:
-				default:
-					break loop
-				}
+		event, err := vmEventQueue.Wait(ctx)
+		if err != nil {
+			if ctx.Err() != nil {
+				return nil
+			} else {
+				klog.Errorf("vmEventQueue returned error: %s", err)
+				return err
 			}
```

Comment:
I believe that you can (should? unclear.) do `errors.Is()` (I also have a `fun/erc.ContextExpired(err)` helper that checks both deadline/canceled cases).

I think, the same thing above: goes for here: Wait will block till the queue is closed or the context is canceled, in either case I think just returning the error (or nil? not sure how this error is processed,) would be _fine_

Reply:
Double-checked, and the caller does a `klog.Fatalf` if the error is non-nil — I think it's better here to explicitly return nil if the context is canceled. I'm going to stick with what's there (roughly) rather than checking via `errors.Is` because I think it's worthwhile to avoid dependence on the reason why the context is done.